File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/http2native/Http2ResponseException.java
package ru.kmorozov.gbd.core.logic.connectors.http2native;

import ru.kmorozov.gbd.core.logic.connectors.ResponseException;

import java.io.IOException;

public class Http2ResponseException extends ResponseException {

    Http2ResponseException(final IOException ioe) {
        super(ioe);
    }

    @Override
    public int getStatusCode() {
        return 500;
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/http2native/Http2ResponseException.kt
package ru.kmorozov.gbd.core.logic.connectors.http2native

import ru.kmorozov.gbd.core.logic.connectors.ResponseException

import java.io.IOException

class Http2ResponseException internal constructor(ioe: IOException) : ResponseException(ioe) {

    override val statusCode: Int
        get() = 500
}
